### ISSUE TITLE
Generated .d.ts file incomplete when multi-entry plugin is used

### ISSUE THREAD

**Issue by rondonjon** [wessberg/rollup-plugin-ts#121]
Hey Frederik, great work ... you should really remove the superflous `.` which breaks the sponsor link in your bug template!

### Versions

- Version:
- Rollup Version: 2.31.0
- Plugin Version: 1.3.5

### Reproduction

rollup.config.js:
``` 
import json from "@rollup/plugin-json";
import multi from "@rollup/plugin-multi-entry";
import ts from "@wessberg/rollup-plugin-ts";
import { terser } from "rollup-plugin-terser";

export default {
  input: "src/**/*.ts",
  output: {
    file: `dist/lib/index.mjs`,
    format: "es"
  },
  plugins: [json(), multi(), terser(), ts()]
};
```

### Expected Behavior

I am working on library project. As depicted above, I am using the multi-entry plugin to (re-) export the named exports of everything in `src/`. I would assume that `rollup-plugin-ts` generates an accompanying `.d.ts` file with all declarations as well.

### Actual Behavior

The generated declaration file contains only a fraction of the types.

I tried a few variations and noticed that the contents of the `.d.ts` file differ when the selection and/or order of the `input:` files change, so I guess that the plugin is either "missing" input files or always overwriting the previously collected declarations from other files.

Removing the `terser()` has *no* effect, i.e. this is probably *not* caused by the minification. Just left it in for completeness.

**Comment by wessberg** [comment]
Hey there. Thanks for the heads up, I'll look into fixing the issue templates. In relation to the bug you reported here, I think I'll need a proper repro to investigate this issue. Please prepare a repro that generates the bad declarations.

The one thing I can say right now is that the order of the Rollup plugins definitely matter, and terser should definitely come after this plugin in all cases (though I did read that removing terser doesn't resolve the issue you describe here).

**Comment by rondonjon** [comment]
Hello @wessberg,

> Please prepare a repro that generates the bad declarations.

Here you go: https://gitlab.com/rondonjon/rollup-plugin-ts-missing-dts

The repo contains two very simple source files, each one exports a type and an instance of that type. After running `yarn build` I have a `dist/index.mjs` file which contains both instances, but the accompanying `index.d.ts` file contains only one of the types (the type from `src/b.js`, which fits to my earlier assumption that only the types from the last processed file make it into the output.
 
> That said, nowadays you can achieve the same thing without using `@rollup/plugin-multi-entry` by passing an array of input files to Rollup

Maybe a misconception or mistake on my side ... just tried it out, but rollup fails with errors and also tries to create multiple output files instead of just 1 with the multi-entry plugin with this approach. Thanks for the hint though, will investigate further.

**Comment by rondonjon** [comment]
PS: possibly unrelated: when I remove the instances and leave only the types in the `src/` files, the generated `index.d.ts` file is completely empty. While this may be a reasonable optimization for applications, it may be problematic for library authors who may want to export type definitions just for external use.

**Comment by rondonjon** [comment]
In the meanwhile: using [dts-generator](https://www.npmjs.com/package/dts-generator) until this gets fixed.

**Comment by rondonjon** [comment]
Hello @wessberg,

just created a `src/index.ts` file in the repo above as follows ...

```
export * from "./a";
export * from "./b";
```

and reduced the entry points to only 1 in the rollup config:

```
import multi from "@rollup/plugin-multi-entry";
import ts from "@wessberg/rollup-plugin-ts";

export default [
  {
    input: "src/index.ts",
    output: {
      file: `dist/lib/index.mjs`,
      format: "es",
    },
    plugins: [multi(), ts()]
  },
];
```

Surprise, problem persists. So the number of entry points seems to be neglectable.

Only after removing `multi()` entirely from `plugins`, I get the expected, "full" d.ts:

```
type A = {
    foo: boolean;
};
declare const a: A;
type B = {
    bar: number;
};
declare const b: B;
export { A, a, B, b };
```

Looks like the problem with the `multi-entry` plugin is more generic, i.e. history gets cleared on every new file and not just for new *entry* files?

Regarding this:

> by passing an array of input files to Rollup

Maybe a misunderstanding. I am neither trying to generate multiple entry points nor multiple chunks in the output bundle, on the contrary, but to bundle an ESM library that has no definite entry point(s) at all and should simply bundle everything from `src/` in one large library JS file.

Maybe I am missing something, but this seems impossible with arrays or objects passed directly to rollup. I tried it out and keep getting multiple output files from rollup instead of just 1 with the `multi-entry` plugin.

**Comment by wessberg** [comment]
I've had some time to dig into this problem.
The problem was that `rollup-plugin-ts` didn't actually understand the new `@rollup/plugin-multi-entry` implementation and expected it to behave like the old `rollup-plugin-multi-entry`. So I've made sure that it is supported now. It has been released as part of [v1.3.7](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.3.7)
